src/python/pants/engine/target_test.py

```diff
@@ -282,33 +279,73 @@ def hydrate_field(
     assert "//:lib" in str(exc)
 
 
+class RustTarget(Target):
+    alias = "rust"
+    core_fields = (Dependencies, Sources)
+
+
+class RustEditionCustomField(StringField):
+    alias = "rust_edition"
+    required = True
+
+
+class FortranCustomField(BoolField):
+    alias = "custom_field"
+    default = False
+
+
+def assert_fortran_target_fields(plugin_fields: RegisteredPluginFields) -> None:
+    assert {FortranExtensions, FortranSources, FortranCustomField} == set(
+        FortranTarget.class_field_types(plugin_fields)
+    )
+
+
+def test_add_custom_fields_bug_last_wins() -> None:
```

Comment:
With Pex, you usually put the bug # in the test name. While we don't normally do that with Pants, I think it would be very helpful here.